docs/source-2.0/guides/model-validation-examples.rst

```diff
@@ -0,0 +1,619 @@
+or service name or within any trait or comment.
+
+.. code-block:: smithy
+
+    {
+        id: "DontUseInternalNamesValidator",
+        name: "ReservedWords",
+        configuration: {
+            reserved: [
+                {
+                    words: ["*spork*"],
+                    reason: """
+                    Sporks are a secret type of silverware. We can't let the rest of the world know about them.
+                    """
+                }
+            ]
+        }
+    }
+
+
+Check that models use a trait at least once
+==================================================
+
+The built-in `EmitNoneSelector` can be used to ensure at least one instance of a trait is found in
+a model. The following example checks that at least one usage of the `@length` trait is found within models
+in the `com.example.weather` namespace.
+
+.. code-block:: smithy
+
+    {
+        "name": "EmitNoneSelector",
+        "id": "NoInstancesOfLengthTrait",
+        "message": "No instances of the length trait were found.",
+        "namespace": ["com.example.weather"]
+        "configuration": {
+            "selector": "[trait|length]"
+        }
+    }
+
+.. note::
+    It is usually concerning if a model does not use common constraint traits such as `@range`, `@pattern`,
+    or `@length` at all
+
+
+Exceptions thrown from operations must have an @httpError trait
+===================================================================
+
+This example shows how to enforce that all errors bound to an operation have an httpError trait (and
+therefore httpError code) defined.
+
+.. code-block:: smithy
+
+    {
+        name: "EmitEachSelector",
+        id: "ErrorHasHttpErrorTrait",
+        configuration: {
+            messageTemplate: """
+            `@{id|name}` is bound as an error but does not have the `@@httpErrorTrait`. \
+            Apply the `@@httpErrorTrait` to this shape.
+            """
+            selector: "operation -[error]-> :not([trait|httpError])"
+        }
+    }
+
+.. note::
+    This example only makes sense for services using an httpProtocol
+
+
+Prefix headers with X-
+========================
+
+Case insensitive check for "X-" prefix on all custom httpHeaders used in your model.
+
+.. code-block:: smithy
+
+    {
+        name: "EmitEachSelector",
+        id: "CustomHeadersHaveXDashPrefix",
+        configuration: {
+            messageTemplate: """
+                @{id|name} applies the `@@httpHeader` with a custom header value of `@{trait|httpHeader}` that does not begin with `x-`.
+                Custom headers should be prefixed with `x-`.
+                """
+            selector: "[trait|httpHeader]:not([trait|httpHeader^='x-' i])"
+        }
+    }
+
+.. note::
+    This example only makes sense for services using an httpProtocol.
+
+.. note::
+    If you prefer to not prefix custom headers with "X-" then you can flip the logic of this validator by changing the
+    selector to `[trait|httpHeader^="x-" i]`
```

Comment:
Same comment as above, any  time you are suggesting that the reader do something, it's _probably_ a tip versus a note.